internal/bminventory/inventory.go

```diff
@@ -1985,6 +2006,57 @@ func optionalParam(data *string, field string, updates map[string]interface{}) {
 	}
 }
 
+func (b *bareMetalInventory) setUsage(enabled bool, name string, props *map[string]interface{}, usages map[string]models.Usage) {
+	if enabled {
+		b.usageApi.Add(usages, name, props)
+	} else {
+		b.usageApi.Remove(usages, name)
+	}
+}
+
+func (b *bareMetalInventory) setDefaultUsage(cluster *models.Cluster) {
```

Comment:
If I understood it correctly - this method(the name is a little bit confusing - it's not the default - it will be different based on the values in the cluster data) is setting all the features usage based on the data we have inside the cluster and we use it only on the register cluster method. Why we can't use it in the updateClusterInternal method as well? why in the update we set each usage separately?

Reply:
Because the updateClusterInternal has more convoluted code for deciding what to update and what to delete (like in case of operators and auto vip)